kv/dist_sender_test.go

```diff
@@ -568,3 +574,64 @@ func TestVerifyPermissions(t *testing.T) {
 	}
 	n.Stop()
 }
+
+// TestSendRPCRetry verifies that sendRPC failed on first address but succeed on
+// second address, the second reply should be successfully returned back.
+func TestSendRPCRetry(t *testing.T) {
+	g := makeTestGossip(t)
+	if err := g.SetNodeDescriptor(&proto.NodeDescriptor{NodeID: 1}); err != nil {
+		t.Fatal(err)
+	}
+	// Fill RangeDescriptor with 2 replicas
+	var descriptor = proto.RangeDescriptor{
+		RaftID:   1,
+		StartKey: proto.Key("a"),
+		EndKey:   proto.Key("z"),
+	}
+	for i := int32(1); i <= 2; i++ {
```

Comment:
Is there a reason you're forcing `int32` here? Looks like you do the appropriate casts to `NodeID` and `StoreID` below, so you should be able to just remove the conversion to `int32`.